server/src/main/java/org/elasticsearch/cluster/metadata/MetadataDataStreamsService.java

```diff
@@ -175,6 +219,36 @@ static ClusterState updateDataLifecycle(
         return ClusterState.builder(currentState).metadata(builder.build()).build();
     }
 
+    /**
+     * Creates an updated cluster state in which the requested data stream has the flag 'rollover_on_write' set to true.
```

Comment:
The flag is passed in, so it's not necessarily always set to true, can you update this javadoc?

Reply:
Good catch! Fixed